app/controllers/posts_controller.rb

```diff
@@ -1,10 +1,28 @@
 class PostsController < ApplicationController
   def index
-    @id = params[:user_id]
+    @user = User.find(params[:user_id])
+    @user_posts = Post.where(author_id: params[:user_id]).all
   end
 
   def show
-    @user_id = params[:user_id]
     @id = params[:id]
+    @user = User.find(params[:user_id])
+    @post_item = Post.where(author_id: @user.id)[params[:id].to_i - 1] if params[:id].to_i
+    @comment = @user.posts[params[:post_id].to_i - 1].comments.new
+  end
+
+  def new
+    @user = current_user
+    @post = @user.posts.new
+  end
+
+  def create
+    @post = Post.create(author: current_user, title: params[:post][:title], text: params[:post][:text],
+                        comments_counter: 0, likes_counter: 0)
+    if @post.new_record?
+      redirect_to '/users/1/posts/new', flash: { wrong: 'Upps! Post was not created.' }
+    else
+      redirect_to '/users/1/posts/new', flash: { success: 'Post was successfully created.' }
+    end
   end
```

Comment:
- [ Optional ] Kindly I recommend you to use `post_params ` as **private method** .This method is used to handle strong parameters in Rails, which is a security feature that prevents assigning request parameters to objects unless they have been explicitly permitted. For more information you can refer this [Link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.) Just like the below sample 

```
private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```